pkg/services/ngalert/migration/models.go

```diff
@@ -42,21 +37,22 @@ type OrgMigration struct {
 
 // newOrgMigration creates a new OrgMigration for the given orgID.
 func (ms *migrationService) newOrgMigration(orgID int64) *OrgMigration {
+	titlededuplicatorPerFolder := make(map[string]*migmodels.Deduplicator)
 	return &OrgMigration{
 		cfg: ms.cfg,
 		log: ms.log.New("orgID", orgID),
 
 		migrationStore:    ms.migrationStore,
 		encryptionService: ms.encryptionService,
 
-		orgID: orgID,
-		// We deduplicate for case-insensitive matching in MySQL-compatible backend flavours because they use case-insensitive collation.
-		seenUIDs:            Deduplicator{set: make(map[string]struct{}), caseInsensitive: ms.migrationStore.CaseInsensitive()},
-		silences:            make([]*pb.MeshSilence, 0),
-		alertRuleTitleDedup: make(map[string]Deduplicator),
-
-		// We deduplicate alert rule groups so that we don't have to ensure that the alerts rules have the same interval.
-		alertRuleGroupDedup: make(map[string]Deduplicator),
+		orgID:    orgID,
+		silences: make([]*pb.MeshSilence, 0),
+		titleDeduplicatorForFolder: func(folderUID string) *migmodels.Deduplicator {
+			if _, ok := titlededuplicatorPerFolder[folderUID]; !ok {
+				titlededuplicatorPerFolder[folderUID] = migmodels.NewDeduplicator(ms.store.GetDialect().SupportEngine(), store.AlertDefinitionMaxTitleLength)
```

Comment:
I think `ms.migrationStore.CaseInsensitive()` was better because it hid the store's implementation details

Reply:
I'll add it back, I don't feel strongly here. Removed it because this was the only place it's used and we have access to the base store.